### ISSUE TITLE
CrossValidationReporter: Display results on each split on the fly

### ISSUE THREAD

**Issue by sylvaincom** [probabl-ai/skore#989]
As suggested by @glemaitre 

### Problem

Suppose I have a long cross-validation on 5 splits and that I made a mistake in my preprocessing. As a result, I get a very bad score on my 5 splits. But I could have saved time because if the score is too bad on my first split, I already know that I made a mistake. No need to wait for the 5 splits to be computed.

### Solution

Display the results of each split progressively, as soon as they are finished, and not only all splits at the end.

When I do cross-validation that takes a few hours, I would like to know as early as possible if there is a problem in my first split so that I can do an early stop.

**Comment by glemaitre** [comment]
Also related to #https://github.com/probabl-ai/skore/issues/949

**Comment by MarieSacksick** [comment]
What do "results" cover here? The metrics?

**Comment by sylvaincom** [comment]
Could be doing a `put` of the estimator report of the first fold?

**Comment by MarieSacksick** [comment]
I don't understand your question; maybe a word is missing?

**Comment by sylvaincom** [comment]
As the cross-validation (let's say `cv=5`) is performed, once the first fold is done, meaning we have done the `fit` on 4 splits and the `predict` on the 5th split, we could store (`put`) the skore estimator report resulting from this first fold. Then, users could visualize the results of the first fold (estimator report of the first fold) on Skore Hub for example: summary of metrics, ROC curve, etc

**Comment by MarieSacksick** [comment]
This seems weird to me, because using skore doesn't mean that people are also using Skore-hub. Also, they may have several projects, etc. So doing a put and creating the CVReport at the same time doesn't seem consistant to me. Maybe it's something that we can do once we removed the put and once we do some sort of auto putting.

**Comment by sylvaincom** [comment]
The `put` could be local (depending on how the project is created), then from another notebook than the one running the cross-validation, users could fetch the results of the stored estimator report of the first fold

If no project is created by the users, we could create a folder called something like "intermediate_results" for safe-keeping purposes

**Comment by auguste-probabl** [comment]
Closing in favor of #3037
